pyro/distributions/empirical.py

```diff
@@ -140,24 +147,20 @@ def event_shape(self):
     def mean(self):
         self._finalize()
         if self._samples.dtype in (torch.int32, torch.int64):
-            raise ValueError("Mean for discrete empirical distribution undefined. Consider converting samples " +
-                             "to ``torch.float32`` or ``torch.float64``.")
-        weights = self._log_weights
-        for _ in range(self._samples.dim() - 1):
-            weights = weights.unsqueeze(-1)
-        return (log_sum_exp(weights, scale=self._samples, dim=0) - log_sum_exp(weights, dim=0)).exp()
+            raise ValueError("Mean for discrete empirical distribution undefined. " +
+                             "Consider converting samples to ``torch.float32`` " +
+                             "or ``torch.float64``.")
```

Comment:
This semantic difference seems large enough that maybe we should consider discrete empirical distributions a separate class?  Thoughts?

Reply:
Are there any methods that you think will be specific to the discrete case? Except for Categorical, I think most other discrete distributions actually return a float tensor, so it will be a limited use case anyways.